mesonbuild/msetup.py

```diff
@@ -174,15 +174,15 @@ def validate_dirs(self, dir1: str, dir2: str, reconfigure: bool, wipe: bool) ->
         return src_dir, build_dir
 
     # See class Backend's 'generate' for comments on capture args and returned dictionary.
-    def generate(self, capture: bool = False, vslite_ctx: dict = None) -> T.Optional[dict]:
+    def generate(self, capture: bool = False, vslite_ctx: T.Optional[dict] = None) -> T.Optional[dict]:
```

Comment:
Can't we type what that dict contains as well?

Reply:
Someone could, certainly, but it is always None except inside the VS backend -- this is specific to the new genvslite backend. And quite frankly I do not want to know how that works or investigate how to type annotate it... I've gone a bit off the whole idea of that backend lol.

And yes, I know that before the genvslite code was merged I went through and added quite a bit of typing to vs2010backend.py